execution/execution.go

```diff
@@ -65,10 +65,10 @@ func New(expr parser.Expr, queryable storage.Queryable, mint, maxt time.Time, st
 		// TODO(fpetkovski): Adjust the step for sub-queries once they are supported.
 		Step: step.Milliseconds(),
 	}
-	return newOperator(expr, selectorPool, opts, hints)
+	return newOperator(expr, selectorPool, opts, hints, extLookbackDelta)
 }
 
-func newOperator(expr parser.Expr, storage *engstore.SelectorPool, opts *query.Options, hints storage.SelectHints) (model.VectorOperator, error) {
+func newOperator(expr parser.Expr, storage *engstore.SelectorPool, opts *query.Options, hints storage.SelectHints, extLookbackDelta time.Duration) (model.VectorOperator, error) {
```

Comment:
Maybe we can add extLookbackDelta to query.Options? This way we don't have to drag it around as another parameter.

Reply:
Good call! Refactored here: https://github.com/thanos-community/promql-engine/pull/216/commits/7408ac13ffbced1d1173fc17c07ad47135e71c6b

Much nicer.